verify.ts

```diff
@@ -0,0 +1,265 @@
+const querystring = require('querystring');
+const abi = require('ethereumjs-abi');
+
+const sleep = (milliseconds: number): Promise<any> => {
+  return new Promise((resolve: () => void): any => setTimeout(resolve, milliseconds));
+};
+
+// tslint:disable: max-line-length
+const optionDefinitions = [
+  { name: 'help', alias: 'h', type: Boolean, description: 'show these command line options' },
+  { name: 'contractAddress', alias: 'a', type: String, description: 'Optional address of a single contract to verify, instead of the default.  Must be given with \'contractName\'.' },
+  { name: 'contractName', alias: 'c', type: String, description: 'Optional name of a single contract to verify, instead of verifying all of them.' },
+  { name: 'network', alias: 'n', defaultOption: true, type: String, description: 'Required name of the network, such as kovan, rinkeby or main.' },
+  { name: 'check', alias: 'g', type: String, description: 'Given verify GUID, check contract verification status.  Ignores other options.' },
+  { name: 'outputFlattened', alias: 'f', type: String, description: 'When verifying, absolute path where to save the flattened .sol file to flattened.["contractName"].sol.' },
+];
+// tslint:enable: max-line-length
+
+const options = commandLineArgs(optionDefinitions);
+
+const usage = (): void => {
+  const sections = [
+    {
+      content: 'Verify migrated Arc contracts to EtherScan.io',
+      header: 'Verify Contracts',
+    },
+    {
+      header: 'Options',
+      optionList: optionDefinitions,
+    },
+  ];
+
+  console.log(commandLineUsage(sections));
+};
+
+const error = (message: string, showUsage: boolean = false) => {
+
+  console.error(message);
+  if (showUsage) {
+    usage();
+  }
+  process.exit(1);
+};
+
+if (options.help) {
+  usage();
+  process.exit(0);
+}
+
+if (!options.network) {
+  error(`'network' is required.  Run 'npm run help' for cli information.`);
+}
+
+if (options.contractAddress && !options.contractName) {
+  error(`contractName is required when contractAddress is given. Run 'npm run help' for cli information.`);
+}
+
+const APIKEY = process.env.APIKEY;
+
+if (!APIKEY) {
+  error(`APIKEY is not found.  See README.md for more information.`);
+}
+
+let addresses = {} as any;
+let allAddresses = {} as any;
+
+if (!options.check) {
+  /**
+   * Enumerate contract names (each with its migrated address) and
+   * fetch the .sol file for it.
+   */
+  const migratedContracts = require('./migration.json');
+
+  allAddresses = (migratedContracts[options.network] && migratedContracts[options.network].base) || undefined;
+  if (!allAddresses) {
+    error(`contracts were not deployed to ${options.network}, or invalid network name`);
+  }
+
+  if (options.contractName && options.contractAddress) {
+    addresses[options.contractName] = options.contractAddress;
+  } else {
+    if (options.contractName) {
+      if (!allAddresses[options.contractName]) {
+        // tslint:disable-next-line: max-line-length
```

Comment:
why to disable that here?

Reply:
i removed it and split the following line.